File: software/examples/java/ExampleSimple.java
import com.tinkerforge.BrickletVoltage;
import com.tinkerforge.IPConnection;

public class ExampleSimple {
	private static final String host = "localhost";
	private static final int port = 4223;
	private static final String UID = "ABC"; // Change to your UID
	
	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the commnents below
	public static void main(String args[]) throws Exception {
		// Create connection to brickd
		IPConnection ipcon = new IPConnection(host, port); // Can throw IOException
		BrickletVoltage vol = new BrickletVoltage(UID); // Create device object

		// Add device to IP connection
		ipcon.addDevice(vol); // Can throw IPConnection.TimeoutException
		// Don't use device before it is added to a connection

		// Get current voltage (unit is mV)
		int voltage = vol.getVoltage(); // Can throw IPConnection.TimeoutException

		System.out.println("Voltage: " + voltage/1000.0 + " V");

		System.console().readLine("Press key to exit\n");
		ipcon.destroy();
	}
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/voltage_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Voltage Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	v, _ := voltage_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current voltage.
	voltage, _ := v.GetVoltage()
	fmt.Printf("Voltage: %f V\n", float64(voltage)/1000.0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
